src/gpu/intel/ocl/gen9_binary.hpp

```diff
@@ -60,7 +60,14 @@ struct gen9_binary_t : public gpu_primitive_t {
                                     && utils::one_of(dst_md()->data_type, f16,
                                             f32, s8, u8, s32))),
                     VERBOSE_UNSUPPORTED_DT);
-            VDISPATCH_BINARY(!is_ternary_op(), VERBOSE_BAD_ALGORITHM);
+            VDISPATCH_BINARY(
+                    (is_ternary_op()
+                            && utils::one_of(src_md(2)->data_type, s8, u8)),
```

Comment:
I think this (and below, and in simple binary) needs to be
```suggestion
                    IMPLICATION(is_ternary_op(),
                            utils::one_of(src_md(2)->data_type, s8, u8)),
```